src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -148,6 +157,11 @@ public void add(String name, List<ByteBuffer> bufferToAdd) throws IOException
 
   public SmooshedWriter addWithSmooshedWriter(final String name, final long size) throws IOException
   {
+    if(useDelegateSmooshWriter)
```

Comment:
Instead of branching off and creating a whole different code path, can you try to figure out how to use the same code path and just have it write to a different location?